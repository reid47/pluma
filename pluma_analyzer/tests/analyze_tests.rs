#[macro_use]
mod macros;

test_analyze! {
  undefined_identifier_simple (false): r#"
    |wat
  "#,

  defined_identifier_simple (true): r#"
    |let fine = 47
    |
    |fine
  "#,

  intrinsic_def_binary_op_valid (true): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + 4
  "#,

  intrinsic_def_binary_op_invalid (false): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + "yikes"
  "#,
}
