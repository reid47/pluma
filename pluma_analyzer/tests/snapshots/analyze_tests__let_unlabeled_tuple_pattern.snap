---
source: pluma_analyzer/tests/analyze_tests.rs
expression: "let tup = (47, \"wow\", 1.23)\nlet (a, b, c) = tup\n\nlet tup2 = (47, (\"wow\", 1.23))\nlet (d, (e, f)) = tup2"
---

=== Source ===
let tup = (47, "wow", 1.23)
let (a, b, c) = tup

let tup2 = (47, ("wow", 1.23))
let (d, (e, f)) = tup2

=== Top-level scope ===
Scope {
    levels: [
        ScopeLevel {
            bindings: [
                (
                    "a",
                    Binding {
                        typ: Int,
                        ref_count: 0,
                        pos: (
                            33,
                            34,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "b",
                    Binding {
                        typ: String,
                        ref_count: 0,
                        pos: (
                            36,
                            37,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "c",
                    Binding {
                        typ: Float,
                        ref_count: 0,
                        pos: (
                            39,
                            40,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "d",
                    Binding {
                        typ: Int,
                        ref_count: 0,
                        pos: (
                            85,
                            86,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "e",
                    Binding {
                        typ: String,
                        ref_count: 0,
                        pos: (
                            89,
                            90,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "f",
                    Binding {
                        typ: Float,
                        ref_count: 0,
                        pos: (
                            92,
                            93,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "tup",
                    Binding {
                        typ: UnlabeledTuple(
                            [
                                Int,
                                String,
                                Float,
                            ],
                        ),
                        ref_count: 1,
                        pos: (
                            4,
                            7,
                        ),
                        kind: Let,
                    },
                ),
                (
                    "tup2",
                    Binding {
                        typ: UnlabeledTuple(
                            [
                                Int,
                                UnlabeledTuple(
                                    [
                                        String,
                                        Float,
                                    ],
                                ),
                            ],
                        ),
                        ref_count: 1,
                        pos: (
                            53,
                            57,
                        ),
                        kind: Let,
                    },
                ),
            ],
        },
    ],
    type_bindings: [],
}

